### ISSUE TITLE
Is there any way to do search for specific logs in the react-lazylog

### ISSUE THREAD

**Issue by leuction** [mozilla-frontend-infra/react-lazylog#12]
Thank you for your fantastic library. I want to do search for logs in this library. Because this library is built with react-virtualized. It seems that I cannot do search with the browser. I have not found any search api in the document. Could you give me a way to do search with that? Thanks.

**Comment by eliperelman** [comment]
@leuction right now the component doesn't expose the raw data, but I think it should in order to do this. Would you be willing to submit a PR to make this possible? For example, I think that the `onLoad` prop should expose the raw response buffer. With that, you could implement a search, and I've done so in the past:

```js
// search

export default (needle, haystack) => {
  const table = [-1, 0];
  const needleLength = needle.length;
  const haystackLength = haystack.length;
  const needleEnd = needleLength - 1;
  let haystackIndex = 0;
  let index = 0;
  let position = 2;
  let needleIndex = 0;

  // Build a table for the search algorithm.
  // This takes O(needleLength) steps.
  while (position < needleLength) {
    if (needle[position - 1] === needle[needleIndex]) {
      needleIndex += 1;
      table[position] = needleIndex;
      position += 1;
    } else if (needleIndex > 0) {
      needleIndex = table[needleIndex];
    } else {
      table[position] = 0;
      position += 1;
    }
  }

  const results = [];

  // Scan the haystack.
  // This takes O(haystackLength) steps.
  while (haystackIndex + index < haystackLength) {
    if (needle[index] === haystack[haystackIndex + index]) {
      if (index === needleEnd) {
        results.push(haystackIndex);
      }

      index += 1;
    } else if (table[index] > -1) {
      haystackIndex = haystackIndex + index - table[index];
      index = table[index];
    } else {
      index = 0;
      haystackIndex += 1;
    }
  }

  return results;
};
```

With the above search function, and if we had the UInt8Array exposed in `onLoad`, you could search with:

```js
onLoad={buffer => {
  const searchPhrase = Array.from(new TextEncoder('utf-8').encode('something'));
  const results = search(searchPhrase, buffer);

  console.log(results);
}}
```

Or that would at least get you a start. Thoughts?

**Comment by ndiaz-candido** [comment]
Hi! I've been working on a search feature for LazyLog. I forked the project and this is the PR in my fork https://github.com/ndiaz-candido/react-lazylog/pull/1

It's still on progress so I didn't want to open a PR to the original project.

I've used eliperelman's algorithm adding something more to get the lines based on the search indexes that the algorithm returns.

**Comment by helfi92** [comment]
Published in v3.3.0. Resolved in https://github.com/mozilla-frontend-infra/react-lazylog/commit/e8951828d6ac4e94d0592b501921de73ced4d990.
